docs: `notmuch address` command updates

ysubach opened the pull request:
- Removed recipients search, default senders search is much faster
- Removed "my@address.org" from command, address seems to be added at the end of line automatically

pazz commented:
Sorry but this is not a "fix": it changes the semantics of the query.
If you remove the trailing address then you can just as well drop fir `from:` alltogether. My original intention was to filter out only those mails that I've sent, because I receive mails from *many* people I don't know and will never reply to.

But I agree that the comment line above your changes could be more elaborate.

ysubach commented:
Oh I see, sorry I completely misunderstood your search query. I'm just getting started with notmuch and alot...

So I made few other changes that I hope make sense now:
* returned original recipients search
* added senders search as an alternative
* added more comments and links to `notmuch address` docs

pazz commented:
no problem and welcome to alot/notmuch :-) 
I'm happy to push these new documentation lines. Please squash them doen into one commit first though.
thanks

ysubach commented:
Ok done

dcbaker commented on docs/source/configuration/contacts_completion.rst:
```diff
@@ -56,18 +56,28 @@ Both respect the `ignorecase` option which defaults to `True` and results in cas
         .. code-block:: ini
 
-           command = 'notmuch address --format=json --output=recipients date:1Y.. AND from:my@address.org'
+           command = 'notmuch address --format=json date:1Y.. AND from:'
```
you're squash is wrong this change should have been removed.

ysubach commented:
Sorry, fixed this commit.

dcbaker commented:
This looks good to me as-is. I'll let @pazz merge if he's happy with it.
